File: test/fixtures/analysis/kotlin-svc/src/main/kotlin/com/example/BackendClient.kt
// Consumed surface: Ktor client member verbs (a $id template segment must
// canonicalize to {var}), plus a runtime-built URL disclosed as dynamic.
package com.example

class BackendClient(private val client: HttpClient) {
    suspend fun item(id: Int): Item = client.get("/api/items/$id").body()

    suspend fun create(item: Item): Item = client.post("/api/items") {
        setBody(item)
    }.body()

    suspend fun opaque(url: String) {
        client.get(url)
    }
}

File: test/fixtures/analysis/csharp-svc/src/Clients/BackendClient.cs
// Consumed surface: HttpClient verb methods (an interpolated URL and a plain
// one), plus a runtime-built URL — recognized and DISCLOSED as dynamic,
// never silently dropped.
namespace Svc.Clients;

public class BackendClient
{
    private readonly HttpClient client;

    public async Task<string?> Fetch(long id)
    {
        return await client.GetFromJsonAsync<string>($"/api/reports/{id}");
    }

    public async Task Submit(HttpContent body)
    {
        await client.PostAsync("/api/reports", body);
    }

    public async Task Opaque()
    {
        await client.GetAsync(BuildUrl());
    }

    private string BuildUrl() => "";
}
